### ISSUE TITLE
Kicking self from space results in broken UI interactions

### ISSUE THREAD

**Issue by kittykat** [vector-im/element-web#19066]
### Steps to reproduce

1. View space members as an Admin of the space
2. Open your own member info in the right panel
3. "Kick" at the bottom of the member info under "Admin"

  First time I kicked myself from "Happy space!" I got taken to this page:
  ![Screenshot from 2021-09-16 07-47-40](https://user-images.githubusercontent.com/51663/133570766-8072ac96-aab0-41fe-ad54-c67e32153478.png)

  I didn't actually get removed from the space after the first kick

4. Refresh, then go back to the member view and "Kick" again
  ![Screenshot from 2021-09-16 07-49-19](https://user-images.githubusercontent.com/51663/133570802-55d67e06-e798-4cb4-a01b-f4abb4cdc4b8.png)

  This time I was kicked from the space, but right panel didn't update to reflect this and I was still left looking at a room in the space.

### What happened?

If I kick myself from a space, I expect to leave the space first time and get taken to the Home view or Join space view.

### URL for webapp

develop.element.io

### Have you submitted a rageshake?

No

**Comment by t3chguy** [comment]
Is this actually specific to Spaces? I think you'd be able to reproduce this with rooms too.

**Comment by t3chguy** [comment]
Should we even expose the Kick & Ban options when you are viewing the member profile on yourself?

**Comment by novocaine** [comment]
We should hide them, I can't think of a sane use case for that.

cc @daniellekirkwood

**Comment by daniellekirkwood** [comment]
Agreed. I can't see why you would want to kick/ban yourself from your own space. 

Let's go ahead and hide the options.
